awx/main/access.py

```diff
@@ -1187,13 +1187,10 @@ def get_value(Class, field):
                 else:
                     return None
 
-        # If a credential is provided, the user should have use access to it.
-        if not self.check_related('credential', Credential, data, role_field='use_role'):
-            return False
-
-        # If a vault credential is provided, the user should have use access to it.
-        if not self.check_related('vault_credential', Credential, data, role_field='use_role'):
-            return False
+        # If credentials is provided, the user should have use access to them.
+        for pk in data.get('credentials', []):
+            if self.user not in get_object_or_400(Credential, pk=pk).use_role:
```

Comment:
Can't create a JT w/ one or more credentials you don't have `use_role` for.

Reply:
you can't create a JT with `credentials` as a distinct API action...

You'd have to create the JT, and then attach the credentials.

We should be able to ignore `credentials` entirely for can_create, can_modify, etc.

but we need to cover `credentials` in `can_attach` and `can_unattach`. That is, unless BaseAccess is sufficient for your purposes, which it is not.

Reply:
Shouldn't that be covered here?

https://github.com/ansible/awx/pull/595/files#diff-c14c9d695bfbeb70889f2b42b18ed42dL1291